client/src/pages/Edit.js

```diff
@@ -16,21 +17,8 @@ function Edit() {
 
   useEffect(() => {
     const editFoodId = async () => {
-      const response = await fetch(`http://localhost:5000/menu_items/${id}`); //change endpoint to hit 1 entry not all
-      const result = {
-        data: null,
-        error: null,
-      };
-
-      if (response.ok) {
-        result.data = await response.json();
-      } else {
-        result.error = await response.text();
-      }
-      console.log(result.data);
-      setEditFood(result.data);
+      queryAPI();
```

Comment:
That's the spot!

Now you'll need to send an `options` argument into this function so it knows what `url` to fetch against, wat `method` to use for the fetch, what do to `onSuccess` and what to do `onError`.

Reply:
<img width="1428" alt="Screen Shot 2023-01-18 at 2 06 44 PM" src="https://user-images.githubusercontent.com/34128735/213272050-df107e27-8363-45e3-9426-1cf98a33e38e.png">

Reply:
Niiiiice 😎